engines/bastion_katello/app/assets/javascripts/bastion_katello/subscriptions/subscriptionAttachAmountFilter.filter.js

```diff
@@ -8,8 +8,16 @@
 angular.module('Bastion.subscriptions').filter('subscriptionAttachAmountFilter',
     ['translate',
     function (translate) {
-        return function (subscription) {
-            var amount = subscription.amount;
+        var amount;
+        var i;
+
+        return function(subscription, mainObject) {
+            var attachedSubscriptions = mainObject.pool_quantities;
+            for (i = 0; i < attachedSubscriptions.length; i++) {
+                if (attachedSubscriptions[i].pool_id === subscription.id) {
+                    amount = attachedSubscriptions[i].quantity;
+                }
+            }
```

Comment:
we typically prefer using underscore js's  `_.each`  function.  Should be a number of examples in the code base